pinot-core/src/main/java/org/apache/pinot/core/query/aggregation/function/DistinctCountThetaSketchAggregationFunction.java

```diff
@@ -108,35 +136,24 @@ public DistinctCountThetaSketchAggregationFunction(List<ExpressionContext> argum
     Preconditions.checkArgument(paramsExpression.getType() == ExpressionContext.Type.LITERAL,
         "Last argument of DistinctCountThetaSketch must be literal (post-aggregation expression)");
     _postAggregationExpression = QueryContextConverterUtils
-        .getFilter(CalciteSqlParser.compileToExpression(postAggregationExpression.getLiteral()));
+        .getExpression(CalciteSqlParser.compileToExpression(postAggregationExpression.getLiteral()));
 
     // Initialize the predicate map
     _predicateInfoMap = new HashMap<>();
-    if (numArguments > 3) {
-      // Predicates are explicitly specified
-      for (int i = 2; i < numArguments - 1; i++) {
-        ExpressionContext predicateExpression = arguments.get(i);
-        Preconditions.checkArgument(predicateExpression.getType() == ExpressionContext.Type.LITERAL,
-            "Third to second last argument of DistinctCountThetaSketch must be literal (predicate expression)");
-        Predicate predicate = getPredicate(predicateExpression.getLiteral());
-        _inputExpressions.add(predicate.getLhs());
-        _predicateInfoMap.put(predicate, new PredicateInfo(predicate));
-      }
-    } else {
-      // Auto-derive predicates from the post-aggregation expression
-      Stack<FilterContext> stack = new Stack<>();
-      stack.push(_postAggregationExpression);
-      while (!stack.isEmpty()) {
-        FilterContext filter = stack.pop();
-        if (filter.getType() == FilterContext.Type.PREDICATE) {
-          Predicate predicate = filter.getPredicate();
-          _inputExpressions.add(predicate.getLhs());
-          _predicateInfoMap.put(predicate, new PredicateInfo(predicate));
-        } else {
-          stack.addAll(filter.getChildren());
-        }
-      }
+
+    // Predicates are explicitly specified
```

Comment:
We should still keep the auto-deriving of predicates. Granted, we won't be able to use $ notation in that case though. What do you think?

Reply:
Hmm I've thought about it, but if we auto-derive like before, we'd only have a subset of features. We won't be able to support SET_DIFF at all if go with that approach right?

Usually, I would think that if there are 2 equivalent ways of writing something, the feature set would be the same. But it seems like only a subset of features would be available if we go with that route.

I'm personally okay with both approach but wanted to point that out.

Reply:
I meant like use the new function call based syntax, with expanded values. For example:
`SET_DIFF('col1=A', 'col2=B')`

Reply:
@mayankshriv I don't quite follow this comment. In what situation do we need the auto-deriving?

Reply:
@bkuang88 I don't understand why we can't support SET_DIFF with auto-derive.

Reply:
> Not related to this PR, but do we want to support theta-sketch without any predicates (union without post aggregation)?

Sure, this can be achieved by making post-aggregation-expression optional as well (or can be empty).

Reply:
Yes, that's what I am stating as well @ManojRThakur. We can auto derive as long as the post-aggr-expression uses  actual predicate strings (as opposed to $ notation).

@Jackie-Jiang Perhaps users will prefer $ notation (in which case you can't auto-derive). But always better to have the option available (especially given that we do support right now).

Reply:
Wonder if that introduces more confusion than flexibility. Because we currently don't want to support complex predicates right? Wouldn't the users make more mistakes or be confused as to when they can use complex predicates and when they cannot?

And if we do start to introduce complex queries in the future, will the syntax be "UNION($1, $2)" or "$1 and $2" or "colA='a' or colB='b'" or "UNION(colA='a', colB='b')"?

Given the vast possibilities out there, I'm wondering if it's better to just stick to a single syntax so that it doesn't cause confusion in the future in case we do want to modify the predicate complexities.

What do you guys think?

Reply:
It seems to me that we can always support this in future (if needed, and/or asked by users) without breaking compatibility with syntax in this PR. So I am fine with not supporting auto-derive for now.